agent/proxycfg/state_test.go

```diff
@@ -798,6 +810,258 @@ func TestState_WatchesAndUpdates(t *testing.T) {
 				},
 			},
 		},
+		"terminating-gateway-initial": testCase{
+			ns: structs.NodeService{
+				Kind:    structs.ServiceKindTerminatingGateway,
+				ID:      "terminating-gateway",
+				Service: "terminating-gateway",
+				Address: "10.0.1.1",
+			},
+			sourceDC: "dc1",
+			stages: []verificationStage{
+				verificationStage{
+					requiredWatches: map[string]verifyWatchRequest{
+						rootsWatchID: genVerifyRootsWatch("dc1"),
+						gatewayServicesWatchID: genVerifyServiceSpecificRequest(gatewayServicesWatchID,
+							"terminating-gateway", "", "dc1", false),
+					},
+					verifySnapshot: func(t testing.TB, snap *ConfigSnapshot) {
+						require.False(t, snap.Valid(), "gateway without root is not valid")
+						require.True(t, snap.ConnectProxy.IsEmpty())
+						require.True(t, snap.MeshGateway.IsEmpty())
+						require.True(t, snap.IngressGateway.IsEmpty())
+					},
+				},
+				verificationStage{
+					events: []cache.UpdateEvent{
+						rootWatchEvent(),
+					},
+					verifySnapshot: func(t testing.TB, snap *ConfigSnapshot) {
+						require.True(t, snap.Valid(), "gateway without services is valid")
+						require.True(t, snap.ConnectProxy.IsEmpty())
+						require.True(t, snap.MeshGateway.IsEmpty())
+						require.True(t, snap.IngressGateway.IsEmpty())
+						require.Equal(t, indexedRoots, snap.Roots)
+						require.Empty(t, snap.TerminatingGateway.WatchedServices)
+						require.Empty(t, snap.TerminatingGateway.ServiceGroups)
+						require.Empty(t, snap.TerminatingGateway.WatchedLeaves)
+						require.Empty(t, snap.TerminatingGateway.ServiceLeaves)
+						require.Empty(t, snap.TerminatingGateway.WatchedResolvers)
+						require.Empty(t, snap.TerminatingGateway.ServiceResolvers)
+						require.Empty(t, snap.TerminatingGateway.WatchedIntentions)
```

Comment:
I think you should be able to replace this with a call to `IsEmpty()`?

Reply:
Yep, fixed. Surfaced that I wasn't checking WatchedResolvers in IsEmpty, so that's fixed now too.